test/unit/test_grouped_loader.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tests for deepreg/dataset/loader/grouped_loader.py in
+pytest style
+"""
+from os.path import join
+
+import numpy as np
+import pytest
+
+from deepreg.dataset.loader.grouped_loader import GroupedDataLoader
+from deepreg.dataset.loader.h5_loader import H5FileLoader
+from deepreg.dataset.loader.nifti_loader import NiftiFileLoader
+
+FileLoaderDict = dict(nifti=NiftiFileLoader, h5=H5FileLoader)
+nifti_path = join("data", "test", "nifti", "grouped")
+h5_path = join("data", "test", "h5", "grouped")
+DataPaths = dict(nifti=nifti_path, h5=h5_path)
+
+
+def test_validate_data_files():
+    """
+    Test validate_data_files function looks for inconsistencies in the fixed/moving image and label lists.
+    If there is any issue it will raise an error, otherwise it returns None.
+    """
+    for key_file_loader, file_loader in FileLoaderDict.items():
+        for train_split in ["train", "test"]:
+            for labeled in ["True", "False", 1, 0]:
+                data_dir_path = join(DataPaths[key_file_loader], train_split)
+                image_shape = (64, 64, 60)
+                common_args = dict(
+                    file_loader=file_loader,
+                    labeled=labeled,
+                    sample_label="all",
+                    intra_group_prob=1,
+                    intra_group_option="forward",
+                    sample_image_in_group=False,
+                    seed=None if train_split == "train" else 0,
+                )
+
+                data_loader = GroupedDataLoader(
+                    data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+                )
+
+                assert data_loader.validate_data_files() is None
+
+
+def test_num_intra_group_probs():
```

Comment:
personally I found the name of this test a bit confusing, maybe name is `test_init`, and then for this case we specify that this is to test single group case?